src/Stratis.Bitcoin/Utilities/DBreezeSerializer.cs

```diff
@@ -124,6 +124,13 @@ internal object Deserializer(byte[] bytes, Type type)
             if (type == typeof(HashHeightPair))
                 return HashHeightPair.Load(bytes);
 
+            if (type == typeof(ProvenBlockHeader))
+            {
+                var provenBlockHeader = new ProvenBlockHeader();
```

Comment:
above a factory is used for block header, is it not necessary here?

Reply:
This required otherwise when a `ProvenBlockHeader` is deserialized from DBreeze.  Otherwise it will throw the following exception :

![image](https://user-images.githubusercontent.com/16000683/46146818-a8868200-c25b-11e8-8faf-ebba6f69dd0f.png)

`BlockHeader` doesn't contain `CoinStake`, `MerkleProof` or `Signature` properties.

Reply:
i was asking about not using the factory , i do understand the case for  PHB is needed

Reply:
I'm assuming when you say PHB you mean `ProvenBlockHeader` ? 

In terms of the factory, this wasn't my implementation - maybe check with @fenix2222.

Reply:
Please check this in the next PR